src/components/Modal/src/ModalLayer.vue

```diff
@@ -114,6 +128,36 @@ export default {
 	right: 0;
 	bottom: 0;
 	left: 0;
+	display: flex;
+	align-items: center;
+	justify-content: center;
+}
+
+.Translucent {
+	position: fixed;
+	top: 0;
+	right: 0;
+	bottom: 0;
+	left: 0;
+	background-color: rgba(0, 0, 0, 0.3);
+}
+
+.ModalContainer {
+	width: 100%;
+	height: 100%;
+	overflow: scroll;
+	box-shadow: 0 0 24px 8px rgba(0, 0, 0, 0.3);
+}
+
+@media screen and (min-width: 1200px) {
+	.ModalContainer {
+		display: inline-block;
+		width: auto;
```

Comment:
We're likely going to need to set a specific width / min-width here. With fluid content everything can be real narrow.  Was testing and you could see the width change with after the image rendered.

Reply:
This is one of those things where I'm not sure whether or not the library should have an opinion about. I think that modals which look "thin" (too narrow vertically or too narrow horizontally) don't look good, and also modals which dynamically resize after being opened also don't look good, and while adding a `min-width` and `min-height` to `.ModalContainer` can help _mitigate_ (not solve) these problems, they're trivially solvable by the end user as long as they simply pass in their own container with a fixed width & height or fixed minimum width & height, e.g.

```vue
<template>
    <m-modal>
        <div style="width: <perfect width for my modal>; height: <perfect height for my modal>;" />
    </m-modal>
</template>
```

Having the end user specify the ideal width & height for their modal seems like a more reliable way to consistently get good looking modals than arbitrarily enforcing some minimum width and height.

And yet just as I finished writing this I realized the above solution won't work for the mobile version of the modal. Hmmm, so either the end user has to be familiar with the breakpoints we use to switch from the mobile version to the desktop version or we add props to the modal like `desktop-width` and `desktop-height`. Actually, the props aren't that great of an idea either, as it would require us to watch the screen using JS. Well, maybe there's a pure CSS solution using custom properties, I'd have to experiment.

Reply:
I hadn't considered that level of flexibility on the width / height, but you make some valid points.  Then it's just the challenges between mobile / desktop. How much CSS styling is too much? (I'll look around at some other super flexible systems for some ideas)

Reply:
Shimaa said width is fixed to 600px and height is dynamic, and switching between different height modals is solved by TransitionResize component